Client/Frontend/Browser/BrowserViewController/BrowserViewController+Wallet.swift

```diff
@@ -407,7 +407,8 @@ extension Tab: BraveWalletEventsListener {
   }
   
   func updateEthereumProperties() {
-    guard let keyringService = BraveWallet.KeyringServiceFactory.get(privateMode: false),
+    guard !isPrivate,
```

Comment:
I believe we need this same Private Browsing check in a few more spots in `BrowserViewController+Wallet.swift`: `emitEthereumEvent`, `accountChangedEvent`, `emitSolanaEvent`, `updateSolanaProperties`.

Reply:
Tested and discussed, only `chainChangedEvent` or `emitEthereumEvent` need the fix applied. This can occur if you open Send in main wallet and change the selected chain.

`accountChangedEvent` is only called when the Solana Provider for the tab is non-nil, but it will be nil in private tab. 
`emitSolanaEvent` is only called from `SolanaProviderScriptHandler` which is only injected in non-private tabs.
`updateSolanaProperties` checks for non-nil Solana Provider in the tab, but it will be nil in private tab.